### PR TITLE
feat(component-testing): Move hmr stuff to the server

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
This was intended to fix the socket hangup issue. AND IT FIXES it. 

## socket hangup fix: 
Here is the appropriate [loc](https://github.com/cypress-io/cypress/blob/93aaee22162cc01e1e04a5aada2b08af406c065e/packages/server-ct/src/routes-ct.ts#L49) which actually resolves the issue. 

The main problem with it was that this is a native node HTTP proxy, and exceptions from there are coming actually from the node itself. Here is a fix, that doing everything

```js
    nodeProxy.web(req, res, {}, (e) => {
      if (e) {
        // eslint-disable-next-line
        console.log('Proxy request error. This is likely the socket hangup issue, we can basically ignore this because the stream will automatically continue once the asset will be available', e)
      }
    }) 
```  
  
 ## moving to the server
 
 Anyway, the fix of socket hangup was easy, but moving HMR out of the client to the server gives us a lot of power:
 1. Allow to pass `hot:false` to webpack. Everything it is doing – creating and running one more socket server that is not needed for us.
 2. Remove frontend connection to the socket (which we don't need) we only need to rerun tests once we get compiled webpack files. 
 3. Still shows the errors nicely with ansi-to-html
 
> This works much much faster than webpack-dev-server + native hot reloading. Because it is not patching each spec with hmr code, it is not opening one more websocket server and it is not listening to the websockets and doing polling on the frontend

### CODE REVIEW

File: packages/server/lib/plugins/dev-server.js [hunk 1]
```diff
@@ -6,11 +6,22 @@ const plugins = require('../plugins')
 
 const baseEmitter = new EE()
 
+let allowToRequest = false
```
Comment (JessicaSachs) [inline packages/server/lib/plugins/dev-server.js:9]: "canRequest" is a bit better "shouldRequest" is also ok

Comment (dmtrKovalenko) [reply packages/server/lib/plugins/dev-server.js:9]: this is not needed, removing right now

Comment (JessicaSachs) [reply packages/server/lib/plugins/dev-server.js:9]: I'm not a fan of using the module scope to init the state... but I understand that the existing model requires plugins.registerHandler to be executed when the file is imported

**Review by JessicaSachs — APPROVED** [review]
LGTM, maybe one rename. The plugins code is nice :-) I'm happy refactored that 👍🏻

**Pull request merged**